### PR TITLE
Implement Python 2 `exec`

### LINKED ISSUE #749: exec not defined on hy2
on hy3 works:

```
$ hy3
hy 0.10.1 using CPython(default) 3.4.2 on Linux
=> (exec "a = 2")
=>
```

on hy2:

```
$ hy2
hy 0.10.1 using CPython(default) 2.7.8 on Linux
=> (exec "a = 2")
Traceback (most recent call last):
  File "<input>", line 1, in <module>
NameError: name 'exec' is not defined
=> 
```

### PR DESCRIPTION
**Pull request by Kodiologist**
Fixes #749.

### CODE REVIEW

**Comment by kirbyfan64** [comment]
My 2¢:

This should be `exec*`, and `exec` itself could be defined conditionally in the core modules, that way semantics are identical for both.

**Comment by Kodiologist** [comment]
@kirbyfan64 Done.

File: hy/core/language.hy [hunk 1]
```diff
@@ -15,7 +15,7 @@
 (if-python2
   (import [StringIO [StringIO]])
   (import [io [StringIO]]))
-(import [hy._compat [long-type]]) ; long for python2, int for python3
+(import [hy._compat [PY3 long-type]])
```
Comment (kirbyfan64) [inline hy/core/language.hy]: `PY3` seems like a randomly added import here...

Comment (Kodiologist) [inline hy/core/language.hy]: Evidently that was some kind of leftover I forgot about. Fixed.

**Comment by Kodiologist** [comment]
@tuturto Poke.

**Pull request merged**